### ISSUE TITLE
Suppress warnings

### ISSUE THREAD

**Issue by danieleds** [simonbrunel/qtpromise#10]
After including the headers, a lot of QtPromise-specific warnings are shown on compilation.

I've been able to suppress them by adding the following pragmas to "QtPromise":

```c++
#ifndef QTPROMISE_MODULE_H
#define QTPROMISE_MODULE_H

#pragma GCC diagnostic push

#pragma GCC diagnostic ignored "-Wunused-local-typedef"
#pragma GCC diagnostic ignored "-Wold-style-cast"

#include "../src/qtpromise/qpromise.h"
#include "../src/qtpromise/qpromisefuture.h"
#include "../src/qtpromise/qpromisehelpers.h"

#pragma GCC diagnostic pop

#endif // ifndef QTPROMISE_MODULE_H
```

It would be nice to not have to do this.

Thanks for the great library.

**Comment by simonbrunel** [comment]
@danieleds I don't have all these warnings, even when compiling with GCC. Where (and how) do you get the `-Wold-style-cast` warnings?  I think I already fixed `-Wunused-local-typedef` in dcbb2ef8609af2ee4dd367d55fb671dc730af631, can you verify if that's the same issue?

**Comment by simonbrunel** [comment]
Fixed in 26a2110, will be released in [version 0.4](/simonbrunel/qtpromise/milestone/3)

**Comment by simonbrunel** [comment]
Released in [0.4.0](/simonbrunel/qtpromise/releases/tag/v0.4.0)
